src/sql/workbench/contrib/notebook/test/browser/notebookService.test.ts

```diff
@@ -85,4 +186,390 @@ suite('Notebook Service Tests', function (): void {
+		const methodName = 'getNavigationProvider';
+		assert.strictEqual(notebookService.getNavigationProvider(), undefined, `${methodName} returns undefined with no providers registered`);
+
+		let providerId = NavigationProviders.NotebooksNavigator;
+		let provider1 = <INavigationProvider>{ providerId: providerId };
+		notebookService.registerNavigationProvider(provider1);
+		let result = notebookService.getNavigationProvider();
+		assert.strictEqual(result, provider1, `${methodName} must return the provider that we registered with netbookService for the provider id: ${provider1.providerId}`);
+
+		providerId = NavigationProviders.ProvidedBooksNavigator;
+		provider1 = <INavigationProvider>{ providerId: providerId };
+		notebookService.registerNavigationProvider(provider1);
+		contextService.createKey(unsavedBooksContextKey, {});
+		result = notebookService.getNavigationProvider();
+
+		assert.strictEqual(result, provider1, `${methodName} must return the provider that we registered with netbookService for the provider id: ${provider1.providerId}`);
+	});
+
+	test('verifies return value of getOrCreateNotebookManager', async () => {
+		await notebookService.registrationComplete;
+		try {
+			await notebookService.getOrCreateNotebookManager(SQL_NOTEBOOK_PROVIDER, undefined);
+		} catch (error) {
+			assert.strictEqual((error as Error).message, NotebookUriNotDefined, `getOrCreateNotebookManager must throw with UriNotDefined error, when a valid uri is not provided`);
+		}
+		const provider1Id = SQL_NOTEBOOK_PROVIDER;
+		const { providerId: provider2Id, manager: provider2Manager } = createRegisteredProviderWithManager({ providerId: 'provider2Id', notebookService });
+
+		const uri1: URI = URI.parse(`untitled:test1`);
+		const uri2: URI = URI.parse(`untitled:test2`);
+		const result1 = await notebookService.getOrCreateNotebookManager(provider1Id, uri1);
+		const result2 = await notebookService.getOrCreateNotebookManager(provider2Id, uri2);
+		const result1Again = await notebookService.getOrCreateNotebookManager(provider1Id, uri1);
+		assert.strictEqual(result2, provider2Manager, `the notebook manager for the provider must be the one returned by getNotebookManager of the provider`);
+		assert.notStrictEqual(result1, result2, `different notebookManagers should be returned for different uris`);
+		assert.strictEqual(result1, result1Again, `same notebookManagers should be returned for same uri for builtin providers`);
+		const result2Again = await notebookService.getOrCreateNotebookManager(provider2Id, uri2);
+		assert.strictEqual(result2, result2Again, `same notebookManagers should be returned for same uri for custom providers`);
+	});
+
+	test('verifies add/remove/find/list/renameNotebookEditor methods and corresponding events', async () => {
+		assert.strictEqual(notebookService.findNotebookEditor(undefined), undefined, `findNotebookEditor should return the editor when searching with the latest Uri`);
```

Comment:
Message seems wrong - you're checking that it's undefined

Reply:
yes will fix message - good catch.

Reply:
fixed.